zerver/lib/subscription_info.py

```diff
@@ -448,6 +466,38 @@ def has_metadata_access_to_previously_subscribed_stream(
     return True
 
 
+def get_setting_values_for_group_settings(
+    group_ids: list[int],
+) -> dict[int, int | AnonymousSettingGroupDict]:
+    setting_value_dict = {}
+    setting_groups = UserGroup.objects.filter(id__in=group_ids).select_related("named_user_group")
+    anonymous_groups = []
+    for group in setting_groups:
+        if hasattr(group, "named_user_group"):
+            setting_value_dict[group.id] = group.id
+        else:
+            setting_value_dict[group.id] = AnonymousSettingGroupDict(
+                direct_members=[],
+                direct_subgroups=[],
+            )
+            anonymous_groups.append(group.id)
+
+    group_members = (
+        UserGroupMembership.objects.filter(user_group_id__in=anonymous_groups)
+        .exclude(user_profile__is_active=False)
+        .values_list("user_group_id", "user_profile_id")
+    )
+    group_subgroups = GroupGroupMembership.objects.filter(
+        supergroup_id__in=anonymous_groups
```

Comment:
I bet it's possible to do a single SQL query to get `UserGroup`, `UserGroupMembership`, and `GroupGroupMembership` with the right joins. This can probably be a follow-up optimization; no code path is impacted so much that it's a big deal, and it should be a drop-in replacement.

If we merge this without resolving this comment, we should file it as a medium-priority goal issue.